src/linux/linux-host-session.vala

```diff
@@ -621,19 +650,43 @@ namespace Frida {
 			return process_name.get_string ();
 		}
 
-		public async void start_activity (string package_name, string? activity_name) throws Error {
+		public async void start_package (string package_name, string method, string? entry_point, string? action) throws Error {
+
 			var package_name_value = new Json.Node.alloc ().init_string (package_name);
 
-			var activity_name_value = new Json.Node.alloc ();
-			if (activity_name != null)
-				activity_name_value.init_string (activity_name);
-			else
-				activity_name_value.init_null ();
+			if (method == "activity") {
 
-			yield call ("startActivity", new Json.Node[] { package_name_value, activity_name_value });
+				var activity_name_value = new Json.Node.alloc ();
+				if (entry_point != null)
+					activity_name_value.init_string (entry_point);
+				else
+					activity_name_value.init_null ();
+
+				yield call ("startActivity", new Json.Node[] { package_name_value, activity_name_value });
+			}
+			else if (method == "receiver") {

```

Comment:
`else if` should go on the same line as `}`.

Reply:
ok